### ISSUE TITLE
Illegal instruction (core dumped)

### ISSUE THREAD

**Issue by martijnmichel** [directus/directus#6307]
Trying out nearly all of the install methods I always get:

root@h2943071:/var/www/vhosts/sonid.app/cms.sonid.app# npm run start

> cms.sonid.app@1.0.0 start /var/www/vhosts/sonid.app/cms.sonid.app
> directus start

Illegal instruction (core dumped)
npm ERR! code ELIFECYCLE
npm ERR! errno 132
npm ERR! cms.sonid.app@1.0.0 start: `directus start`
npm ERR! Exit status 132
npm ERR! 
npm ERR! Failed at the cms.sonid.app@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-06-16T08_17_57_903Z-debug.log

**Comment by martijnmichel** [comment]
when using docker:
/usr/local/bin/entrypoint: line 5:     7 Illegal instruction     (core dumped) npx directus bootstrap

**Comment by paescuj** [comment]
What's your environment?

**Comment by martijnmichel** [comment]
I am installing it on a strato dedicated server with Ubuntu 20.04 LTS 64bit + Plesk Obsidian

Server specs at https://www.strato.nl/server/dedicated-server-linux/
AMD Opteron 4180
6 x 2,6 GHz
16 GB RAM
2 x 1 TB HDD

**Comment by paescuj** [comment]
Thanks! Might be due to AMD processor or Plesk doing some magic. Something of interest in the `/root/.npm/_logs/2021-06-16T08_17_57_903Z-debug.log` file?

**Comment by martijnmichel** [comment]
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'start' ]
2 info using npm@6.14.13
3 info using node@v14.17.1
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle cms.sonid.app@1.0.0~prestart: cms.sonid.app@1.0.0
6 info lifecycle cms.sonid.app@1.0.0~start: cms.sonid.app@1.0.0
7 verbose lifecycle cms.sonid.app@1.0.0~start: unsafe-perm in lifecycle true
8 verbose lifecycle cms.sonid.app@1.0.0~start: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/var/www/vhosts/sonid.app/cms.sonid.app/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
9 verbose lifecycle cms.sonid.app@1.0.0~start: CWD: /var/www/vhosts/sonid.app/cms.sonid.app
10 silly lifecycle cms.sonid.app@1.0.0~start: Args: [ '-c', 'directus start' ]
11 silly lifecycle cms.sonid.app@1.0.0~start: Returned: code: 132  signal: null
12 info lifecycle cms.sonid.app@1.0.0~start: Failed to exec start script
13 verbose stack Error: cms.sonid.app@1.0.0 start: `directus start`
13 verbose stack Exit status 132
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:375:28)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:375:28)
13 verbose stack     at maybeClose (internal/child_process.js:1055:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
14 verbose pkgid cms.sonid.app@1.0.0
15 verbose cwd /var/www/vhosts/sonid.app/cms.sonid.app
16 verbose Linux 5.4.0-74-generic
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "start"
18 verbose node v14.17.1
19 verbose npm  v6.14.13
20 error code ELIFECYCLE
21 error errno 132
22 error cms.sonid.app@1.0.0 start: `directus start`
22 error Exit status 132
23 error Failed at the cms.sonid.app@1.0.0 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 132, true ]

**Comment by martijnmichel** [comment]
root@h2943071:~/directus# docker run -p 8055:8055 directus/directus
13:40:23 ⚠️  >
13:40:23 ⚠️  >                         WARNING!
13:40:23 ⚠️  >
13:40:23 ⚠️  >  The KEY and SECRET environment variables are not set. Some
13:40:23 ⚠️  >  temporary variables were generated to fill the gap, but in
13:40:23 ⚠️  >        production this is going to cause problems.
13:40:23 ⚠️  >
13:40:23 ⚠️  >                        Reference:
13:40:23 ⚠️  > https://docs.directus.io/reference/environment-variables.html
13:40:23 ⚠️  >
13:40:23 ⚠️  >
/usr/local/bin/entrypoint: line 5:    28 Illegal instruction     (core dumped) npx directus bootstrap

**Comment by rijkvanzanten** [comment]
Seeing the error message `Illegal instruction (core dumped)`, this is most likely a compatibility problem in the specific software/hardware stack of this host.

I'll close this for now, as I don't think this is something we can address in Directus itself, but we can keep debugging 🙂

**Comment by martijnmichel** [comment]
Please help me out, I really need directus, it was running fine on the VPS from the same provider :(
What info do you need?

**Comment by benhaynes** [comment]
> Please help me out, I really need directus, it was running fine on the VPS from the same provider :(
> What info do you need?

Hey, apologies that things aren't going smoothly for you. Unfortunately our core team can't debug server/stack issues. With millions of users, this is just untenable for our small team. You can try to reach out to our community:

https://directus.chat

**Comment by inta** [comment]
Ok, I have also come across this problem. Since RC 70 onwards im getting the error:
> | /usr/bin/entrypoint: line 5:     8 Illegal instruction     (core dumped) npx directus bootstrap

when run inside Docker. RC 59, 67, 68, 69 are working without any problem.

The test machine is a VM (qemu) with Debian 11 as guest OS. The CPU hardware is AMD but the VM itself has a kvm64 virtual CPU. I might test it again on an Intel based platform in the future, but currently I have none.

I tried to install directus directly on the same VM using `npx create-directus-project foobar` with the following result:
> Illegal instruction
> ✖ Installing Directus
> Error: undefined

Both ways end successful on my local dev machine which is also powered by an AMD CPU (and Debian 11), so I think an hardware issue is not that likely.

Is there some change in RC 70 that comes to your mind which could cause problems like this?

**Comment by rijkvanzanten** [comment]
@inta No idea! Must be a dependency with a native component that refuses to build under that specific environment..

**Comment by Schwenniii** [comment]
i have the same issue with run `npx create-directus-project my-project`

Illegal instruction
✖ Installing Directus
Error: undefined

**Comment by m4rrc0** [comment]
Same for me on dedicated server: Intel Atom N2800 (2c/4t - 1.86GHz) / 4GB / 2TB / Ubuntu 21
Tried the CLI installation and manual but `init` and `bootstrap` give the same error.

The only thing I installed on that server before directus is node (through the version manager n).

**Comment by azrikahar** [comment]
@m4rrc0 Thank you very much for looking into it. Admittedly I don't really know much in that aspect, but I figured I should at least honor the effort you had put in. 

I believe I might have narrowed down the actual culprit between RC69 and RC70. If we trace back on the commits _before_ RC70: https://github.com/directus/directus/commits/a6a354a0ae336ae77a430dfaab7224a68dbfd14e, we can see that Sharp was upgraded from `0.27.2` to `0.28.3` here: https://github.com/directus/directus/pull/5924

Pairing this newfound knowledge with the error `Illegal instruction (core dumped)`, I found this answer by the owner of Sharp package specifically on this exact issue: https://github.com/lovell/sharp/issues/2723#issuecomment-842956537

Based on his answer, seems like the solution for this would be to compile your own libvips. Once again thanks for the in depth investigation from your side, and I hope this does clear things up now 👍

**Comment by m4rrc0** [comment]
Why am I under the impression than when something is broken in a web dev stack, it is always because of Sharp?! 😄 

Thanks a lot for catching the ball so quickly. I will probably be able to try this today. I'll report back when it is done.

**Comment by m4rrc0** [comment]
So... I feel like it is going to be messy if one wants to persevere on that road.

Any version of Directus 9 (after RCs) requires Sharp 0.29 which apparently needs at least libvips 8.11.3 (or does not garantee it will work with older versions?) which is not 'old enough'.
I tried installing libvips 8.10.5 globally from source. I think I was successful (because `pkg-config --modversion vips-cpp` outputs `8.10.5`) but after running `npm i` on Directus, `directus bootstrap` still outputs `Illegal instruction`.

I may very well have missed something when installing from source. I am not sure how to check if sharp properly used my libvips 8.10.5.

Another approach:
Trying to force sharp@0.27.2 on Directus 9 (with `npm-force-resolutions`) throws an error: `Cannot find module 'color'`. It may be fixable but to be honest I am not comfortable messing with dependencies just to make it work.

My journey will end there as I installed Directus on another server anyway. Sorry I couldn't help more.
